src/gc-alloc-profiler.cpp

```diff
@@ -268,18 +87,30 @@ RawBacktrace get_raw_backtrace() {
 // == exported interface ==
 
 JL_DLLEXPORT void jl_start_alloc_profile(int skip_every) {
-    jl_printf(JL_STDERR, "g_alloc_profile se:%d allocs:%d \n", g_alloc_profile.skip_every, g_alloc_profile.allocs.size());
     g_alloc_profile_enabled = true;
-    g_alloc_profile.reset(skip_every);
+    g_alloc_profile = AllocProfile{skip_every};
 }
 
 extern "C" {  // Needed since the function doesn't take any arguments.
 
-JL_DLLEXPORT struct AllocResults jl_stop_and_write_alloc_profile() {
+JL_DLLEXPORT struct RawAllocResults jl_stop_alloc_profile() {
     g_alloc_profile_enabled = false;
 
-    return AllocResults{g_alloc_profile.allocs.size(),
-                    g_alloc_profile.allocs.data()};
+    return RawAllocResults{
+        g_alloc_profile.allocs.size(),
+        g_alloc_profile.allocs.data()
+    };
+}
+
+JL_DLLEXPORT void jl_free_alloc_profile() {
```

Comment:
@NHDaly I wrote this in case people don't like the C++ destructors… 🤷‍♂️

Reply:
Alas, too bad. But yeah, I think that makes sense. It's definitely worse, IMO, but it's more standard for julia's C/C++ style. 🤷